### PR TITLE
Night Vision Goggles update

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This is a reboot of #84550 but with way less unbearable: mesons are off the table and I've made the night vision tonalities lighter now, while also improving the darkness cutoffs of some of them. Btw, you can also turn night vision googles on/off now. Each pair of departmental night vision goggles now comes with their own tint to boot.

As always, this doesn't affect nukies and ninjas. They're almost garaunteed to keep their googles equipped and on at all time, so it would honestly adulterate the overall experience a little.

Now, for some pictures...

In the dark, **top** row is **old**, **bottom** row is **new**. from left to right: security, science, standard/meson, diagnostic, health:
dank

In the light: security, science, no goggles, standard/meson, diagnostic, health:
lite

The screenshots were taken from varedited goggles on the servers because it's bit faster than booting up a local server, plus I had to finetune a few things.

## Why It's Good For The Game
So, I kinda wanted to give a bit of screen colors to goggles that was not some sort of "optional, by default turned off" kind of crap nobody cares about, however my previous PR was met with a lot of criticism, which made me rethink how to do it. I honestly don't want it to be a burden, but I do want to make them a bit fancier tho.

Also the the medical NVs (and perhaps sec NVs) were a bit crappier than the rest.

## Changelog

:cl:
add: Night vision goggles now subtly tint your screen.
add: You can turn your night vision goggles off. Doing so removes the tint and the eye protection malus.
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
-1 unless the NVGs make the splinter cell "phweeeeeee" sfx when turned on

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -174,8 +186,18 @@
...
+	glass_colour_type = /datum/client_colour/glass_colour/lightgreen
+	actions_types = list(/datum/action/item_action/toggle_nv)
+	forced_glass_color = TRUE
+
+/obj/item/clothing/glasses/night/update_icon_state()
+	. = ..()
+	icon_state = length(color_cutoffs) ? initial(icon_state) : "night_off"
+
+/obj/item/clothing/glasses/night/colorless
+	desc = "You can totally see in the dark now! Now with 50% less green!"
```
Comment (MrMelbert) [inline code/modules/clothing/glasses/_glasses.dm:199]:
```suggestion
	desc = parent_type::desc + " Now with 50% less green!"
```

**Comment by Ghommie** [comment]
Made the darkness cutoff of the medical hud less intense:
immagine

**Comment by Ghommie** [comment]
> -1 unless the NVGs make the splinter cell "phweeeeeee" sfx when turned on

Pass me the sfx then.

**Comment by MrMelbert** [comment]
https://freesound.org/people/Syna-Max/sounds/60345/

**Comment by Metekillot** [comment]
I cannot tell any difference except perhaps an immensely slight one with diagnostics.


Toggleable is nice! Maybe nukies could get ones that swap between NVGs and sunglasses so forgetting to game with your goggle toggle means you flukie on your first flashbang

**Comment by Metekillot** [comment]
Ohh there's no tint now and now there is ok 

Why did you post 'old' versions, that was confusing for me  <:)

**Comment by Ghommie** [comment]
> I cannot tell any difference except perhaps an immensely slight one with diagnostics.

Any more than this and some people would start to complain. Not everyone sees colors the same.

> Toggleable is nice! Maybe nukies could get ones that swap between NVGs and sunglasses so forgetting to game with your goggle toggle means you flukie on your first flashbang

Nukies already wear helmets that negate the drawbacks of NVGs. That said, don't be near a flashbang when it detonates.

**Comment by SimplySmiley123** [comment]
> > I cannot tell any difference except perhaps an immensely slight one with diagnostics.
> 
> Any more than this and some people would start to complain. Not everyone sees colors the same.
> 
> > Toggleable is nice! Maybe nukies could get ones that swap between NVGs and sunglasses so forgetting to game with your goggle toggle means you flukie on your first flashbang
> 
> Nukies already wear helmets that negate the drawback of NVGs. That said, don't be near a flashbang when it detonates.

To be entirely honest the tint not being optional just makes me want to use night vision goggles even less.
Nobody used the optional tints on _any_ eyepiece because, let's be honest, it's just super annoying and makes it way harder to see things for nothing but flavor.

**Comment by Ghommie** [comment]
> To be entirely honest the tint not being optional just makes me want to use night vision goggles even less.

**_Even_** less? Gee, if you don't want to use night vision googles to begin with, is it really my problem? 🙄

**Comment by projectkepler-ru** [comment]
> > To be entirely honest the tint not being optional just makes me want to use night vision goggles even less.
> 
> **_Even_** less? Gee, if you don't want to use night vision googles to begin with, is it really my problem? 🙄

if you made security helmet sprite looks like uncircumised foreskin and then made it the only way you can get pepperspray protection

and suddenly no one wants to wear it

Yes
it would be your problem

**Comment by Ghommie** [comment]
Thanks for the mental image, but I do believe your example to be a bit hyperbolic and flawed, since it's not quite what I'm doing.

**Comment by Wallemations** [comment]
Smh CLEARLY night vision should blow your eyes out with brightness when you turn lights on smh smh smh smh
So unrealistic smh

**Pull request merged**